### PR TITLE
feat(data-grid): port data grid component from fast

### LINKED ISSUE #993: Port FAST data-grid

### PR DESCRIPTION
**Pull request by YonatanKra**
Closes #993

### CODE REVIEW

File: apps/docs/_data/components.json [hunk 1]
```diff
@@ -308,6 +308,15 @@
 		"modules": [
 			"/assets/modules/components/combobox/index.js",
 			"/assets/modules/components/option/index.js"
+		]
+	},
+	{
+		"title": "data-grid",
+		"markdown": "./libs/components/src/lib/data-grid/README.md",
```
Comment (yinonov) [inline apps/docs/_data/components.json:315]:
```suggestion
		"status": "alpha",
		"markdown": "./libs/components/src/lib/data-grid/README.md",
```

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/data-grid/README.md [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+<vwc-data-grid></vwc-data-grid>
+<script>
+    // TODO:: add example
+    const grid = document.querySelector('vwc-data-grid');
+    grid.rowsData = [
+        {data1: 'data11', data2: 'data12'},
+        {data1: 'data21', data2: 'data22'},
+    ];
+</script>
+```
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:98]: is this the correct snippet? there's no `rowItemTemplate`

File: libs/components/src/lib/data-grid/README.md [hunk 3]
```diff
@@ -0,0 +1,235 @@
...
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:117]: correct snippet? there's no `cellItemTemplate`

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/data-grid/README.md [hunk 4]
```diff
@@ -0,0 +1,259 @@
...
+        {data1: 'data31', data2: 'data32'},
+        {data1: 'data41', data2: 'data42'},
+        {data1: 'data51', data2: 'data52'},
+        {data1: 'data61', data2: 'data62'},
+    ];
+</script>
+```
+
+
+### noTabbing
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:46]: why not simply `tabindex="-1"`?

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/data-grid/README.md [hunk 5]
```diff
@@ -0,0 +1,259 @@
...
+    grid.rowsData = [
+        {data1: 'data11', data2: 'data12'},
+        {data1: 'data21', data2: 'data22'},
+    ];
+</script>
+```
+
+## Members
+### Generate-header
+Use `generate-header"` for data gris deader
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:22]:
```suggestion
Use `generate-header"` for data grid deader
```

File: libs/components/src/lib/data-grid/README.md [hunk 6]
```diff
@@ -0,0 +1,259 @@
+# data-grid
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:1]: documentation must be more elaborative, I can hardly understand what the members stand for

File: libs/components/src/lib/data-grid/README.md [hunk 7]
```diff
@@ -0,0 +1,259 @@
...
+    ];
+</script>
+```
+
+### columnDefinitions
+
+- Type: `ColumnDefinition[]`
+- Default: `null`
+
+The column definitions of the grid
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:88]: which is?

**Review by yinonov — CHANGES_REQUESTED** [review]
I tend to start reviewing from reading spec from the readme representation. It's hard when the docs aren't clear. I can obviously try reading unit test instead but consumers will first try docs

File: libs/components/src/lib/data-grid/README.md (hunk 4 above)
Comment (YonatanKra) [reply libs/components/src/lib/data-grid/README.md:56]: It has a more complex behavior internally IMO.

File: libs/components/src/lib/data-grid/README.md [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+    ];
+</script>
+```
+
+## Members
+### Generate-header
+Use `generate-header"` for data grid header visibility mode.
+
+- Type: `none` | `default`| `sticky`
+- Default: `default`
```
Comment (yinonov) [inline libs/components/src/lib/data-grid/README.md:25]: I know this comes from extending _**fast**_ but this is a very strange API. default it the fallback of set of available values. it should not by an option on its own

**Comment by yinonov** [comment]
notice vonage/vivid#1371

File: libs/components/src/lib/data-grid/README.md (hunk 8 above)
Comment (YonatanKra) [reply libs/components/src/lib/data-grid/README.md:25]: Default = show the header
Sticky = show a sticky header
none = do not show the header
Do you want to change the `default` value?

**Review by yinonov — APPROVED** [review]

**Pull request merged**